amm-executor/src/Spectrum/Executor/EventSink/Handlers/Orders.hs

```diff
@@ -40,41 +40,71 @@ import Spectrum.Executor.EventSource.Data.TxContext
   ( TxCtx(LedgerCtx) )
 import Spectrum.Executor.Backlog.Persistence.BacklogStore
   ( BacklogStore (BacklogStore, get) )
+import Spectrum.LedgerSync.Config (NetworkParameters(NetworkParameters, systemStart))
+import Cardano.Api (SlotNo(unSlotNo))
+import Cardano.Slotting.Time (SystemStart(getSystemStart))
+import Spectrum.Executor.Backlog.Config (BacklogServiceConfig(BacklogServiceConfig, orderLifetime))
+import System.Logging.Hlog (Logging (Logging, infoM))
 
 mkPendingOrdersHandler
   :: MonadIO m
   => WriteTopic m (OrderInState 'Pending)
+  -> QSem
+  -> Logging m
+  -> BacklogServiceConfig
+  -> NetworkParameters
   -> EventHandler m 'LedgerCtx
-mkPendingOrdersHandler WriteTopic{..} = \case
-  AppliedTx (MinimalLedgerTx MinimalConfirmedTx{..}) ->
-    foldl process (pure Nothing) (txOutputs <&> parseOrder)
+mkPendingOrdersHandler WriteTopic{..} syncSem logging@Logging{..} BacklogServiceConfig{..} NetworkParameters{..} = \case
+  AppliedTx (MinimalLedgerTx MinimalConfirmedTx{..}) -> do
+    currentTime <- getCurrentTime
+    let
+       slotsTime = secondsToNominalDiffTime . fromIntegral $ unSlotNo slotNo
+       txTime    = addUTCTime slotsTime (getSystemStart systemStart)
+    if (diffUTCTime currentTime txTime > orderLifetime)
+      then infoM ("Tx is outdated : " ++ show txId) >> pure Nothing
+      else (parseOrder logging `traverse` txOutputs) >>= foldM (process txTime) Nothing           
       where
-        process _ ordM = do
-          ts <- getCurrentTime
-          mapM publish $ ordM <&> flip PendingOrder ts
+        process oTime _ ordM = (\order -> liftIO (signalQSem syncSem) >> publish order) `mapM` (ordM <&> flip PendingOrder oTime)
```

Comment:
Let's not use `traverse` and `mapM` in infix notation as they are not supposed to be used like that.

Reply:
fixed